src/clr-core/tag/tag.element.spec.ts

```diff
@@ -5,27 +5,132 @@
  */
 import { CdsTag } from '@clr/core/tag';
 import '@clr/core/tag';
-import { componentIsStable, createTestElement, removeTestElement, waitForComponent } from '@clr/core/test/utils';
+import {
+  componentIsStable,
+  createTestElement,
+  getComponentSelector,
+  removeTestElement,
+  waitForComponent,
+} from '@clr/core/test/utils';
+
+const getTagComponentSelector = getComponentSelector('cds-tag');
+function classnameFromString(str: string) {
+  if (str.length > 0) {
+    return '.' + str;
+  }
+  return '';
+}
 
 describe('tag element', () => {
   let testElement: HTMLElement;
-  let component: CdsTag;
   const placeholderText = 'Tag Placeholder';
+  const defaultClassname = 'clickable-tag';
+  const readonlyClassname = 'readonly-tag';
+  const closableClassname = 'closable-tag';
+
+  const closeIconString = '<cds-icon shape="times"';
+
+  function getTagComponentFromElement(el: HTMLElement, type: 'default' | 'readonly' | 'closable' = 'default'): CdsTag {
+    let classname: string;
+    switch (type) {
+      case 'readonly':
+        classname = readonlyClassname;
+        break;
+      case 'closable':
+        classname = closableClassname;
+        break;
+      default:
+        classname = defaultClassname;
+    }
+    return el.querySelector<CdsTag>(getTagComponentSelector(classnameFromString(classname)));
+  }
```

Comment:
This seems a bit complicated, would this be functionally equivalent?

```typescript
function getTagComponentFromElement(el: HTMLElement, type: 'readonly' | 'closable' | 'clickable' = 'clickable') {
  return el.querySelector<CdsTag>(`cds-tag.${type}-tag`);
}
```

Reply:
It sure could. I'll make that happen.

Reply:
✅